### ISSUE TITLE
nuxt/client.js Module build failed SyntaxError: Invalid regular expression flags

### ISSUE THREAD

**Issue by esanmar** [gjovanov/facer#11]
Hi
Congratulations for your great application!
I have instaled it in a shared hosting an I got this error

Thanks, regards

**Comment by esanmar** [comment]
ERROR in ./.nuxt/client.js
Module build failed (from ./node_modules/eslint-loader/dist/cjs.js):
/usr/home/ofercita/www/facer/node_modules/eslint/lib/source-code/source-code.js:426
        return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
                                       ^

SyntaxError: Invalid regular expression flags
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:607:28)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/home/ofercita/www/facer/node_modules/eslint/lib/source-code/index.js:4:17)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)

**Comment by gjovanov** [comment]
Hi @esanmar ,
can you pls elaborate on what specifically did u change and during what command is this error happening?

**Comment by esanmar** [comment]
Hi Goran,

I have changed nuxt.config.js HOST and /www/facer/node_modules/eslint/lib/source-code/source-code.js

/*return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));*/
return /\s/u.test(text.replace(/\/\*.*?\*\//, "gus"));	

But I still got errors:

10 silly lifecycle facer@1.0.0~api: Args: [ '-c', 'cross-env NODE_ENV=development node api/start.js' ]
11 info lifecycle facer@1.0.0~api: Failed to exec api script
12 verbose stack Error: facer@1.0.0 api: `cross-env NODE_ENV=development node api/start.js`
12 verbose stack spawn ENOENT
12 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:48:18)
12 verbose stack     at emitTwo (events.js:126:13)
12 verbose stack     at ChildProcess.emit (events.js:214:7)
12 verbose stack     at maybeClose (internal/child_process.js:925:16)
12 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
13 verbose pkgid facer@1.0.0
14 verbose cwd /usr/home/ofercita/www/facer
15 verbose Linux 3.16.0-10-amd64
16 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "api"
17 verbose node v8.9.4
18 verbose npm  v5.6.0
19 error file sh
20 error code ELIFECYCLE
21 error errno ENOENT
22 error syscall spawn
23 error facer@1.0.0 api: `cross-env NODE_ENV=development node api/start.js`
23 error spawn ENOENT
24 error Failed at the facer@1.0.0 api script.
24 error This is probably not a problem with npm. There is likely additional logging output above.
25 verbose exit [ 1, true ]


Thanks!

**Comment by gjovanov** [comment]
Sorry, I don't understand, why would you change the source code of eslint?
- /www/facer/node_modules/eslint/lib/source-code/source-code.js

As the error says:
`SyntaxError: Invalid regular expression flags`

Your regular expression has wrong syntax.

Again, why would you change eslint source code?

**Comment by esanmar** [comment]
Because before I changed I got this error:

http://ofercita.com/error2.png

**Comment by esanmar** [comment]
Because before I changed I got this error:

![error2](https://user-images.githubusercontent.com/14100142/70689180-d23a9600-1cb3-11ea-972e-0352458b5712.png)

**Comment by gjovanov** [comment]
What are the exact steps performed until you reach this error?

E.g.
1 `git clone https://github.com/gjovanov/facer.git`
2. `npm i`
3. `npm run api`
4. `npm run dev` or `npm run build` ?

or something else is that you are doing?


Also pls try to revert you changes and try to build without any change of your, even without the nuxt.config.js and see if you get the same error...

**Comment by esanmar** [comment]
Hi,

I have made the 4 points, npm run build.
I see these errors, do I have to create a package.json?

![error1](https://user-images.githubusercontent.com/14100142/70736320-5d497980-1d10-11ea-8116-f97fe4112d6f.PNG)
![error2](https://user-images.githubusercontent.com/14100142/70736317-5d497980-1d10-11ea-8e63-4abc4aacb48c.PNG)

[2019-12-12T17_50_34_874Z-debug.log](https://github.com/gjovanov/facer/files/3957149/2019-12-12T17_50_34_874Z-debug.log)

**Comment by gjovanov** [comment]
`npm i` needs to be run inside the folder `facer`. It installs all packages based on the `facer/package.json`. I suggest you read a bit nodeJS apps and NPM (node package manager) to understand the basic, because without that, this app would be way too advanced to explain in github issue.

So, once again, these are the steps:
1 `git clone https://github.com/gjovanov/facer.git` (this will download the app in a folder called `facer`)
2. `cd facer` (to enter inside the `facer` folder)
2. `npm i` (to install the packages based on the package.json)
3. `npm run api` (to start API - backend  - server)
4. `npm run build` (to build the UI - frontend
5. `npm run start` (to start the UI server)

After that your UI server is listening on port 3000 and the backend API on 3001. You can change this in nuxt.config.js later. First make sure you succeed the mentioned 5 steps.

**Comment by esanmar** [comment]
Sorry but I have not experince in this enviroment.
The point 5 shows this error, with the original nuxt.config.js.
After the error I have changed the nuxt.config.js because of is a shared hosting and localhost is not accesible, but the error still shows:

![error](https://user-images.githubusercontent.com/14100142/70751226-3b5fef00-1d30-11ea-8073-e0f565cba053.PNG)

Thank you for your time
[2019-12-12T21_29_27_211Z-debug.log](https://github.com/gjovanov/facer/files/3957935/2019-12-12T21_29_27_211Z-debug.log)

**Comment by gjovanov** [comment]
The error message is not very telly, so it's hard to give u a direction about the root cause.

However, what you can do is two things:
1. Instead of `npm run build` and after that `npm run start`, you can try to run in Development by only running `npm run dev`. But still have the api server started before by `npm run api`. See if you encounter the same error.
2. Additionally, you can do this on your local machine Laptop/PC all 5 steps from above and see if it works on your local machine.

My assumption is that the shared hosting you are using has some constraints that prevents the app from starting. Basically, when the app starts it uses the package `pm2` which basically starts as many processes of the app as the number of Cores your CPU has. And it might be that your user account is not allowed (it's constrained) to start processes on other Cores of the CPU. And if this is the case, then you need a better hosting e.g. your own managed  server, for example have a look at (hetzner.de)[https://www.hetzner.de/] where you have pretty cheap options for that.

To summarize, to dismiss or confirm by assumption about the shared hosting constraints, pls run it locally on your Laptop/PC and see if get the same error.

The prerequisite is that you will have to have installed:
- nodejs
- git
to do that.


Let me know the outcome of this.

**Comment by gjovanov** [comment]
Actually, there is third option you can try out.

You can edit the file `facer/package.json` and add the following line in `scripts` section in between the `"start"` script and `"generate"` script:

`"start-simple": "cross-env NODE_ENV=production node ./server/index.js",`

this line introduces a new script that allows you to be able to start the app on e single Core of CPU.

After you have edited the `package.json` and save the changes, then you can run:

`npm run start-simple`

Keep in mind this is a replacement for the normal `npm run start`, so all steps need to be followed as above except that the script `start` was replaced by the script `start-simple.`


Let me know how it goes for you.

**Comment by esanmar** [comment]
I have added the line in the package.json but the run buld shows an error:



Can I install it in Windows 10, I have installed node and I have downloaded the  facer folder from github

**Comment by esanmar** [comment]
I have added the line facer/package.json and in It works, but when I try to upload an image shows this error:

![errorupload](https://user-images.githubusercontent.com/14100142/70887071-412e2c80-1fdd-11ea-8dde-fdf0c55f1242.png)

I think is better to install it in local, I have Windows 10, and I have installed node.exe and I have downloaded the folder facer from github.
The instalattion steps are the same in Windows?

Thanks

**Comment by gjovanov** [comment]
Yes the steps are the same on WIN 10 as well.

Regarding the error you encountered on the hosting server, what node and npm versions do you have installed on it? You can check this by `node -v` and `npm -v`. Pls make sure you have the newest versions installed.

Also on the hosting server, you can cleanup all the dependencies and reinstall them from scratch by:
- `rm -rf node_modules`,
- then `npm i`


Also why is it showing API listening on http://ofercita.com:3000 ? Did you change something in the configs? As per default config, it should listen on localhost:3000 as per default

**Comment by gjovanov** [comment]
Any update on this?

**Comment by esanmar** [comment]
I am trying to install it on Windows Server 2016 with no success.
I get these errors:


![e1](https://user-images.githubusercontent.com/14100142/71002501-ae29eb00-20df-11ea-90ac-a61820df2830.png)

![e2](https://user-images.githubusercontent.com/14100142/71002500-ae29eb00-20df-11ea-997a-1063a641324b.png)

[2019-12-17T10_31_27_279Z-debug.log](https://github.com/gjovanov/facer/files/3973655/2019-12-17T10_31_27_279Z-debug.log)

**Comment by esanmar** [comment]
Do I have to execute this?

npm config set strict-ssl false

Thanks

**Comment by gjovanov** [comment]
Well, that might help, but it might be that you either have an old version of NodeJS or NPM.

It seems that while the packagge `sharp` is being downloaded, the URL contains self-assigned certificate, which by default is not allowed. So then it falls back to manually build `sharp` from the source code, for which u need to have [win-build-tools](https://www.npmjs.com/package/windows-build-tools) installed. But again, I wouldn't bother with that, the normal download of sharp works on WIN. So you might have old Node or NPM.

On my laptop, I have these and it works fine:
`node -v`
v12.2.0

`npm -v`
6.9.0

if you have newer version, it should be a problem, only if u have older you would need to upgrade.

Try both:
1) `npm config set strict-ssl false`
2) `node -v` and `npm -v` - upgrade if necessary

**Comment by esanmar** [comment]
node -v --> V12.13.1

npm -v shows this error:

![er1](https://user-images.githubusercontent.com/14100142/71067316-a5372900-2174-11ea-93eb-7dc4cd197d72.PNG)

Then I go to the Repair option in Windows and solved, this is the version:

npm -v  --> 6.12.1

But when I execute npm i, removes de npm folder and I get the error again.

It seems that this is the problem, npm i removes the npm folder.

Do I have to skip this step?

Thanks

**Comment by gjovanov** [comment]
Something is terrible wrong here @esanmar. Npm should be automatically available as a global package and not as a local package in `facer/node_modules`.

I assume your PATH variable for NPM is not set properly or if you installed node recently, then you might just need to close CMD and reopen again or run `refreshenv` to refresh the ENV variables.

Pls position yourself  in any other folder e.g. `cd C:` and from there run `npm -v`. If it yields result it means that NPM is available globally.

Usually NPM get's installed automatically with the installation of NodeJS. How do you install NodeJS?

This seems totally out of scope and has nothing to do with `facer`, but the basic understanding of NodeJS and NPM.

I seriosly suggest you look for online tutorials on NPM and Node, because that knowledge is a prerequisite for this project. You need a couple of hours. But you still to invest the time.

**Comment by gjovanov** [comment]
Hi @esanmar ,
you are right there were two issues that are now fixed if you pull the latest version.

1. `npm run start` wasn't working because `pm2` was missing from the `package.js` so when you restored the packages by running `npm i`, pm2 was missed. This happened because recently I made the effort to upgrade all dependencies/packages to their latest version. Anyways, now its fixed
2. Since the DB of users and user images has grown during the train process, it was failing because the nodejs has a default limit of 5MB JSON that you can post via a request. Now it's increase to 50MB. But keep in mind even this can become too little in case you have too many users you are trying to recognize.

Let me know if you can start the app now.

**Comment by gjovanov** [comment]
Any update on this one @esanmar ?

Or we can close the issue?

**Comment by gjovanov** [comment]
Closing due to inactivity.

Feel free to reopen with further details.
